packages/web-media-box/src/pipelines/mse/mseManager.ts

```diff
@@ -0,0 +1,178 @@
+import Logger from "@/utils/logger";
+import type { SourceBufferOperation, SourceBufferWrapper } from "./types/bufferOperation";
+import { OperationType } from "./consts/sourceBuffer";
+
+export default class MseManager {
+  private readonly logger: Logger;
+  private mediaSource: MediaSource;
+  private sourceBuffers: Map<string, SourceBufferWrapper>;
+  private sourceOpen: Promise<void>;
+
+  public constructor(logger: Logger) {
+    this.logger = logger.createSubLogger('MseManager');
+    //TODO: ManagedMediaSource
+    this.mediaSource = new MediaSource();
+    this.sourceOpen = this.initMediaSource();
+    this.sourceBuffers = new Map();
+  }
+
+  public initBuffers(mimeCodecs: Array<string>) {
+    mimeCodecs.forEach((mimeCodec) => {
+      if (MediaSource.isTypeSupported(mimeCodec)) {
+        const mimeType = mimeCodec.substring(0, mimeCodec.indexOf(';'));
```

Comment:
as far as I remember, we want to pass container + codec string to the `addSourceBuffer `

Reply:
Good catch, missed this while refactoring for the operations queue. Fix incoming.